packages/apps-engine/src/server/ProxiedApp.ts

```diff
@@ -53,11 +53,6 @@ export class ProxiedApp {
     public async call(method: `${AppMethod}`, ...args: Array<any>): Promise<any> {
         let options;
 
-        // Pre events need to be fast as they block the user
-        if (method.startsWith('checkPre') || method.startsWith('executePre')) {
```

Comment:
This is probably the only part i don't like. If PRE callbacks block the use, if a user registers a `executePreMessageSent` that takes the full timeout, it would take 30s by default.

For normal handlers that's good but for these, i wonder if they should have their own env, and maybe a smaller default value.

Reply:
That's a valid concern. The use case that prompted this change was the usage of an antivirus solution, like the ClamAV app, where the time it takes for a file to be analyzed will increase with the file size.

What we may want to do is cut short the timeout of pre-message events only, which are extremelly critical, where file uploads can be expected to take a while longer

Reply:
I like that idea

Reply:
It might be a dumb idea, but why not do something like:

1. Users upload the file
2. On `PreFileUpload` schedule a job, and replace the file with a custom message "File is getting analyzed"
3. Once the file is done analyzing, update the default message with the File if it passed, if not show any error/delete.

Reply:
@Dnouv that wouldn't work. On `IPreFileUpload`, you don't get to change the text of the message that will be sent, that happens after the upload has been accepted.

Reply:
@KevLehman I'd like to leave that idea for another time, though. Maybe it is a non-problem that we're trying to fix.